src/dpapimk_fmt_plug.c

```diff
@@ -0,0 +1,503 @@
+/* DPAPI masterkey file version 1 and 2 cracker by
+ * Fist0urs <jean-christophe.delaunay at synacktiv.com>
+ *
+ * This software is Copyright (c) 2017 
+ * Fist0urs <jean-christophe.delaunay at synacktiv.com>,
+ * and it is hereby released to the general public under the following terms:
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted.
+ *
+ * All credits for the algorithm go to "dpapick" project,
+ * https://bitbucket.org/jmichel/dpapick
+ * and Dhiru Kholia <dhiru.kholia at gmail.com> for his
+ * work on the DPAPI masterkey file version 1 implementation.
+ */
+
+#if FMT_EXTERNS_H
+extern struct fmt_main fmt_DPAPImk;
+#elif FMT_REGISTERS_H
+john_register_one(&fmt_DPAPImk);
+#else
+
+#include <string.h>
+#include <assert.h>
+#include <errno.h>
+#include <openssl/des.h>
+
+#ifdef _OPENMP
+#include <omp.h>
+#ifndef OMP_SCALE
+#define OMP_SCALE               64
+#endif
+#endif
+
+#include "arch.h"
+#include "misc.h"
+#include "memory.h"
+#include "common.h"
+#include "formats.h"
+#include "params.h"
+#include "options.h"
+#include "unicode.h"
+#include "aes.h"
+#include "sha.h"
+#include "md4.h"
+#include "hmac_sha.h"
+#include "memdbg.h"
+
+#define DPAPI_CRAP_LOGIC
+
+#include "pbkdf2_hmac_sha512.h"
+#include "pbkdf2_hmac_sha1.h"
+
+#define FORMAT_LABEL            "DPAPImk"
+#define FORMAT_TAG              "$DPAPImk$"
+#define FORMAT_TAG_LEN          (sizeof(FORMAT_TAG) - 1)
+#define FORMAT_NAME             "DPAPI masterkey file v1 and v2"
+
+#if defined(SIMD_COEF_64) && defined(SIMD_COEF_32)
+#define ALGORITHM_NAME          "SHA1/MD4 PBKDF2-(SHA1/SHA512)-DPAPI-variant 3DES/AES256 " SHA1_ALGORITHM_NAME
+#else
+#define ALGORITHM_NAME          "SHA1/MD4 PBKDF2-(SHA1/SHA512)-DPAPI-variant 3DES/AES256 32/" ARCH_BITS_STR
+#endif
+
+#define BENCHMARK_COMMENT       ""
+#define BENCHMARK_LENGTH        -1
+#define BINARY_SIZE             0
+#define PLAINTEXT_LENGTH        125
+#define MAX_CT_LEN              4096
+#define MAX_SID_LEN             1024
+#define SALT_SIZE               sizeof(*cur_salt)
+#define BINARY_ALIGN            1
+#define SALT_ALIGN              sizeof(int)
+
+#define MAX_IV_LEN 16
+
+#define KEY_LEN2 32
+#define IV_LEN2 16
+#define DIGEST_LEN2 64
+
+#define KEY_LEN1 24
+#define IV_LEN1 8
+#define DIGEST_LEN1 20
+
```

Comment:
Nitpick: The values of these define statements can be aligned to match the existing alignment. It might look better :-)